omod/src/main/webapp/resources/scripts/htmlForm.js

```diff
@@ -258,24 +278,50 @@
     // TODO: these methods (getEncounter*Date*) will have to be modified when/if we switch datepickers
     // TODO: could/should be generalized so as not to be datepicker dependent?
 
-    htmlForm.setEncounterStartDateRange = function(date) {
+    //Set encounterDate datepicker start date
+    htmlForm.setEncounterStartDateRange = function(date , handletimezones) {
         if (getField('encounterDate.value')) {
-            getField('encounterDate.value').datepicker('option', 'minDate', date);
+            if (handletimezones) {
+                var startDateClientTimezone = new Date(date)
+                getField('encounterDate.value').datepicker('option', 'minDate', startDateClientTimezone)
+            }else{
+                getField('encounterDate.value').datepicker('option', 'minDate',  extractDate(date));
+            }
         }
     };
 
-    htmlForm.setEncounterStopDateRange = function(date) {
+    //Set encounterDate datepicker stop date
+    htmlForm.setEncounterStopDateRange = function(date, handletimezones) {
+        //Set browser timezone
+        if(jq("#encounterDate").find(".hfe-timezone")){
+            jq("#encounterDate").find(".hfe-timezone").val(Intl.DateTimeFormat().resolvedOptions().timeZone)
+        }
         if (getField('encounterDate.value')) {
-            getField('encounterDate.value').datepicker('option', 'maxDate', date);
+            if (handletimezones) {
+                var stopDateClientTimezone = new Date(date)
+                getField('encounterDate.value').datepicker('option', 'maxDate', stopDateClientTimezone > new Date ? new Date : stopDateClientTimezone)
```

Comment:
Do you mind explaining a bit in regards to the use of this logic? `stopDateClientTimezone > new Date ? new Date : stopDateClientTimezone`

Reply:
What i wanted to do is: If stopDate with clientTimezone is after the "new Date" (the now time) then is in the future, but the stopdate cannot be on the future, so in this case, we use `new Date`

Reply:
Is that automatically overwriting the user's value instead of giving them an error message?

Reply:
The stopDate is not a value that the users give. It is a value that the backend gets from the visit. That logic was already implemented on the back-end I just grab it to the front-end.

Reply:
Oh i see!